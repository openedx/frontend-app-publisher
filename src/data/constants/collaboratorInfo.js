export const CREATE_COLLABORATOR = 'CREATE_COLLABORATOR';
export const CREATE_COLLABORATOR_SUCCESS = 'CREATE_COLLABORATOR_SUCCESS';
export const CREATE_COLLABORATOR_FAIL = 'CREATE_COLLABORATOR_FAIL';
export const EDIT_COLLABORATOR_INFO = 'EDIT_COLLABORATOR_INFO';
export const EDIT_COLLABORATOR_INFO_SUCCESS = 'EDIT_COLLABORATOR_INFO_SUCCESS';
export const EDIT_COLLABORATOR_INFO_FAIL = 'EDIT_COLLABORATOR_INFO_FAIL';
export const CANCEL_COLLABORATOR_INFO = 'CANCEL_COLLABORATOR_INFO';
export const RESET_COLLABORATOR_INFO = 'RESET_COLLABORATOR_INFO';
