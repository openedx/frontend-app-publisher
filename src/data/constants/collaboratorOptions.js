export const REQUEST_COLLABORATOR_OPTIONS = 'REQUEST_COLLABORATOR_OPTIONS';
export const REQUEST_COLLABORATOR_OPTIONS_FAIL = 'REQUEST_COLLABORATOR_OPTIONS_FAIL';
export const REQUEST_COLLABORATOR_OPTIONS_SUCCESS = 'REQUEST_COLLABORATOR_OPTIONS_SUCCESS';
