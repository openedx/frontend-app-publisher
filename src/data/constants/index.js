const VERIFIED_TRACK = Object.freeze({ name: 'Verified and Audit', key: 'verified' });
const AUDIT_TRACK = Object.freeze({ name: 'Audit Only', key: 'audit' });
const PROFESSIONAL_TRACK = Object.freeze({ name: 'Professional Only', key: 'professional' });
const MASTERS_TRACK = Object.freeze({ key: 'masters' });
const ENTITLEMENT_TRACKS = Object.freeze(['verified', 'professional']);

const REVIEW_BY_LEGAL = 'review_by_legal';
const REVIEW_BY_INTERNAL = 'review_by_internal';
const PUBLISHED = 'published';
const REVIEWED = 'reviewed';
const UNPUBLISHED = 'unpublished';
const ARCHIVED = 'archived';
const IN_REVIEW_STATUS = Object.freeze([REVIEW_BY_LEGAL, REVIEW_BY_INTERNAL]);
const POST_REVIEW_STATUSES = Object.freeze([PUBLISHED, REVIEWED]);
const COURSE_EXEMPT_FIELDS = Object.freeze(['imgSrc', 'videoSrc']);
const COURSE_RUN_NON_EXEMPT_FIELDS = Object.freeze(['expected_program_name', 'expected_program_type']);

const DATE_FORMAT = 'YYYY-MM-DD[T]HH:mm:ss[Z]';
const DATE_INPUT_PATTERN = '20[1-9][0-9]/(0[1-9]|1[012])/(0[1-9]|[12][0-9]|3[01])';
const FORMAT_DATE_MATCHER = /20\d{2}-(0\d{1}|1[0-2])-([0-2]\d{1}|3[0-1])/;
const NORMALIZE_DATE_MATCHER = /20\d{2}\/(0\d{1}|1[0-2])\/([0-2]\d{1}|3[0-1])/;

const EXECUTIVE_EDUCATION_SLUG = 'executive-education-2u';
const BOOTCAMP_SLUG = 'bootcamp-2u';

const COURSE_URL_SLUG_PATTERN_OLD = '^[a-z0-9_]+(?:-[a-z0-9_]+)*$';
const COURSE_URL_SLUG_PATTERN_NEW = '^learn/[a-z0-9_]+(?:-?[a-z0-9_]+)*/[a-z0-9_]+(?:-?[a-z0-9_]+)*$';
const COURSE_URL_SLUG_PATTERN = `${COURSE_URL_SLUG_PATTERN_NEW}|${COURSE_URL_SLUG_PATTERN_OLD}`;

const COURSE_URL_SLUG_VALIDATION_MESSAGE = {
  [COURSE_URL_SLUG_PATTERN_OLD]: 'Course URL slug contains lowercase letters, numbers, underscores, and dashes only.',
  [COURSE_URL_SLUG_PATTERN_NEW]: 'Course URL slug contains lowercase letters, numbers, underscores, and dashes only and must be in the format learn/<primary_subject>/<org-slug>-<course_slug>',
  [COURSE_URL_SLUG_PATTERN]: 'Course URL slug contains lowercase letters, numbers, underscores, and dashes only and must be in the format <custom-url-slug> or learn/<primary_subject>/<org-slug>-<course_slug>.',
};

export {
  VERIFIED_TRACK,
  AUDIT_TRACK,
  PROFESSIONAL_TRACK,
  MASTERS_TRACK,
  ENTITLEMENT_TRACKS,
  REVIEW_BY_LEGAL,
  REVIEW_BY_INTERNAL,
  PUBLISHED,
  REVIEWED,
  UNPUBLISHED,
  ARCHIVED,
  IN_REVIEW_STATUS,
  POST_REVIEW_STATUSES,
  DATE_FORMAT,
  DATE_INPUT_PATTERN,
  FORMAT_DATE_MATCHER,
  NORMALIZE_DATE_MATCHER,
  COURSE_EXEMPT_FIELDS,
  COURSE_RUN_NON_EXEMPT_FIELDS,
  EXECUTIVE_EDUCATION_SLUG,
  BOOTCAMP_SLUG,
  COURSE_URL_SLUG_PATTERN,
  COURSE_URL_SLUG_PATTERN_NEW,
  COURSE_URL_SLUG_PATTERN_OLD,
  COURSE_URL_SLUG_VALIDATION_MESSAGE,
};
