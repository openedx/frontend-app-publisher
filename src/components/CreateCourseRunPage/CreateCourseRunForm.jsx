import moment from 'moment';
import React from 'react';
import { Field, reduxForm } from 'redux-form';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import ButtonToolbar from '../ButtonToolbar';
import FieldLabel from '../FieldLabel';
import ActionButton from '../ActionButton';

import {
  endDateHelp, runTypeHelp, startDateHelp, pacingHelp, keyHelp,
} from '../../helpText';
import RenderInputTextField from '../RenderInputTextField';
import RenderSelectField from '../RenderSelectField';
import DateTimeField from '../DateTimeField';
import {
  getDateWithDashes, getOptionsData, isSafari, localTimeZone, parseOptions,
} from '../../utils';
import { DATE_INPUT_PATTERN } from '../../data/constants';

const BaseCreateCourseRunForm = ({
  handleSubmit,
  pristine,
  isCreating,
  title,
  uuid,
  courseRunOptions,
  courseRunTypeOptions,
  currentFormValues,
  courseRunLabels,
  canSetRunKey,
}) => {
  const courseRunOptionsData = getOptionsData(courseRunOptions);
  const { pacing_type: { choices } } = courseRunOptionsData;
  const pacingTypeOptions = courseRunOptionsData && parseOptions(choices);
  return (
    <div className="create-course-run-form">
      <h2>Create a new course run</h2>
      <hr />
      <FieldLabel text="Course" className="mb-2" />
      <div className="mb-3">
        <Link to={`/courses/${uuid}`}>
          {title}
        </Link>
      </div>
      <form onSubmit={handleSubmit}>
        <Field
          name="rerun"
          type="text"
          component={RenderSelectField}
          options={courseRunLabels}
          label={(
            <FieldLabel
              id="rerun.label"
              text="Select a run to copy"
              helpText={(
                <div>
                  <p>
                    Select a run to copy as the starting point for your new studio instance.
                    The newest run is listed first.
                  </p>
                </div>
              )}
            />
          )}
        />
        {canSetRunKey
        && (
        <Field
          name="courseRunKey"
          component={RenderInputTextField}
          type="text"
          pattern="[a-zA-Z0-9-]+"
          label={(
            <FieldLabel
              id="courseRunKey-label"
              text="Run Key"
              helpText={keyHelp}
              optional
            />
          )}
        />
        )}
        {/* TODO this should be refactored when paragon supports safari */}
        {/* text inputs for safari */}
        {isSafari
          ? (
            <div>
              <Field
                name="start"
                type="text"
                component={DateTimeField}
                dateLabel="Start date"
                timeLabel={`Start time (${localTimeZone})`}
                helpText={startDateHelp}
                required
                maxLength="10"
                pattern={DATE_INPUT_PATTERN}
                placeholder="yyyy/mm/dd"
              />
              <Field
                name="end"
                type="text"
                component={DateTimeField}
                dateLabel="End date"
                timeLabel={`End time (${localTimeZone})`}
                helpText={endDateHelp}
                required
                maxLength="10"
                pattern={DATE_INPUT_PATTERN}
                placeholder="yyyy/mm/dd"
              />
            </div>
          )
          // date inputs for all browsers besides safari
          : (
            <div>
              <Field
                name="start"
                type="date"
                component={DateTimeField}
                dateLabel="Start date"
                timeLabel={`Start time (${localTimeZone})`}
                helpText={startDateHelp}
                required
                minDate={getDateWithDashes(moment())}
              />
              <Field
                name="end"
                type="date"
                component={DateTimeField}
                dateLabel="End date"
                timeLabel={`End time (${localTimeZone})`}
                helpText={endDateHelp}
                required
                minDate={getDateWithDashes(moment(currentFormValues.start).add(1, 'd') || moment())}
              />
            </div>
          )}
        <Field
          name="run_type"
          component={RenderSelectField}
          options={courseRunTypeOptions}
          label={(
            <FieldLabel
              id="run_type.label"
              text="Course run enrollment track"
              required
              helpText={runTypeHelp}
            />
          )}
          required
        />
        <Field
          name="pacing_type"
          type="text"
          component={RenderSelectField}
          options={pacingTypeOptions}
          label={(
            <FieldLabel
              id="pacing_type.label"
              text="Course pacing"
              helpText={pacingHelp}
            />
          )}
        />
        <ButtonToolbar>
          <Link to={`/courses/${uuid}`}>
            <button
              type="button"
              className="btn btn-outline-primary"
              disabled={isCreating}
            >
              Cancel
            </button>
          </Link>
          <ActionButton
            disabled={pristine}
            labels={{
              default: 'Create',
              pending: 'Creating',
            }}
            state={isCreating ? 'pending' : 'default'}
          />
        </ButtonToolbar>
      </form>
    </div>
  );
};

BaseCreateCourseRunForm.defaultProps = {
  currentFormValues: {},
  courseRunLabels: [],
  courseRunOptions: {},
  courseRunTypeOptions: [],
};

BaseCreateCourseRunForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  uuid: PropTypes.string.isRequired,
  pristine: PropTypes.bool.isRequired,
  isCreating: PropTypes.bool.isRequired,
  currentFormValues: PropTypes.shape({
    start: PropTypes.string,
  }),
  courseRunLabels: PropTypes.arrayOf(PropTypes.shape({})),
  courseRunOptions: PropTypes.shape({
    data: PropTypes.shape(),
    error: PropTypes.arrayOf(PropTypes.string),
    isFetching: PropTypes.bool,
  }),
  courseRunTypeOptions: PropTypes.arrayOf(PropTypes.shape({})),
  canSetRunKey: PropTypes.bool.isRequired,
};

const CreateCourseRunForm = reduxForm({
  form: 'create-course-run-form',
})(BaseCreateCourseRunForm);
export { BaseCreateCourseRunForm, CreateCourseRunForm };
