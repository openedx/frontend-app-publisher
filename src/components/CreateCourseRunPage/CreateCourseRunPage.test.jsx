import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Alert } from '@edx/paragon';

import { CreateCourseRunForm } from './CreateCourseRunForm';
import CreateCourseRunPage from './index';
import { courseOptions } from '../../data/constants/testData';

describe('CreateCourseRunPage', () => {
  it('renders html correctly', () => {
    const component = shallow(<CreateCourseRunPage
      id="00000000-0000-0000-0000-000000000001"
      courseInfo={{
        data: {},
        isFetching: false,
        isCreating: false,
        error: null,
      }}
      courseOptions={courseOptions}
    />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
  it('renders html correctly with Course Type', () => {
    const component = shallow(<CreateCourseRunPage
      id="00000000-0000-0000-0000-000000000001"
      courseInfo={{
        data: {
          type: '8a8f30e1-23ce-4ed3-a361-1325c656b67b',
        },
        isFetching: false,
        isCreating: false,
        error: null,
      }}
      courseOptions={courseOptions}
    />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
  it('renders html correctly when fetching', () => {
    const component = shallow(<CreateCourseRunPage
      id="00000000-0000-0000-0000-000000000001"
      courseInfo={{
        data: {},
        isFetching: true,
        isCreating: false,
        error: null,
      }}
    />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
  it('renders html correctly when creating', () => {
    const component = shallow(<CreateCourseRunPage
      id="00000000-0000-0000-0000-000000000001"
      courseInfo={{
        data: {},
        isFetching: false,
        isCreating: true,
        error: null,
      }}
    />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
  it('renders html correctly when error', () => {
    const component = shallow(<CreateCourseRunPage
      id="00000000-0000-0000-0000-000000000001"
      courseInfo={{
        data: {},
        isFetching: false,
        isCreating: false,
        error: ['failed'],
      }}
    />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('refuses access to form when course is under review', () => {
    const component = shallow(<CreateCourseRunPage
      id="00000000-0000-0000-0000-000000000001"
      courseInfo={{
        data: {
          course_runs: [{
            status: 'review_by_legal',
            key: 'course-v1:edX+cs101+2T2019',
          }],
          title: 'Test Course',
        },
        isFetching: false,
        isCreating: false,
        error: null,
      }}
    />);

    // Confirm message is shown
    const reviewAlert = component.find(Alert);
    const reviewMessage = 'Test Course has been submitted for review. No course runs can be added right now.';
    expect(reviewAlert.text()).toEqual(reviewMessage);

    // And confirm that we don't show form
    const form = component.find(CreateCourseRunForm);
    expect(form).toHaveLength(0);
  });
});
