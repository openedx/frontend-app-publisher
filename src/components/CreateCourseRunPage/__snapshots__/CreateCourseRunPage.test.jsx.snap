// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateCourseRunPage renders html correctly 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    Create Course Run
  </HelmetWrapper>
  <PageContainer
    className=""
    wide={false}
  >
    <div>
      <ReduxForm
        courseOptions={
          Object {
            "data": Object {
              "actions": Object {
                "POST": Object {
                  "level_type": Object {
                    "choices": Array [
                      Object {
                        "display_name": "Beginner",
                        "value": "beginner",
                      },
                      Object {
                        "display_name": "Intermediate",
                        "value": "intermediate",
                      },
                      Object {
                        "display_name": "Advanced",
                        "value": "advanced",
                      },
                    ],
                  },
                  "subjects": Object {
                    "child": Object {
                      "choices": Array [
                        Object {
                          "display_name": "Business",
                          "value": "business",
                        },
                        Object {
                          "display_name": "Chemistry",
                          "value": "chemistry",
                        },
                        Object {
                          "display_name": "English",
                          "value": "english",
                        },
                        Object {
                          "display_name": "Security",
                          "value": "security",
                        },
                      ],
                    },
                  },
                  "type": Object {
                    "type_options": Array [
                      Object {
                        "course_run_types": Array [
                          Object {
                            "is_marketable": true,
                            "modes": Array [
                              "credit",
                              "verified",
                              "audit",
                            ],
                            "name": "Credit",
                            "uuid": "f17e29d6-4648-4bb5-a199-97dc40f904aa",
                          },
                          Object {
                            "is_marketable": true,
                            "modes": Array [
                              "verified",
                              "audit",
                            ],
                            "name": "Verified and Audit",
                            "uuid": "4e260c57-24ef-46c1-9a0d-5ec3a30f6b0c",
                          },
                          Object {
                            "is_marketable": true,
                            "modes": Array [
                              "audit",
                            ],
                            "name": "Audit Only",
                            "uuid": "cfacfc62-54bd-4e1b-939a-5a94f12fbd8d",
                          },
                        ],
                        "entitlement_types": Array [
                          "verified",
                        ],
                        "name": "Credit",
                        "tracks": Array [
                          Object {
                            "mode": Object {
                              "certificate_type": "credit",
                              "is_credit_eligible": true,
                              "is_id_verified": true,
                              "name": "Credit",
                              "payee": "platform",
                              "slug": "credit",
                            },
                            "seat_type": Object {
                              "name": "Credit",
                              "slug": "credit",
                            },
                          },
                          Object {
                            "mode": Object {
                              "certificate_type": "verified",
                              "is_credit_eligible": false,
                              "is_id_verified": true,
                              "name": "Verified",
                              "payee": "platform",
                              "slug": "verified",
                            },
                            "seat_type": Object {
                              "name": "Verified",
                              "slug": "verified",
                            },
                          },
                          Object {
                            "mode": Object {
                              "certificate_type": "",
                              "is_credit_eligible": false,
                              "is_id_verified": false,
                              "name": "Audit",
                              "payee": "",
                              "slug": "audit",
                            },
                            "seat_type": Object {
                              "name": "Audit",
                              "slug": "audit",
                            },
                          },
                        ],
                        "uuid": "9521aa7d-801b-4a67-92c3-716ea30f5086",
                      },
                      Object {
                        "course_run_types": Array [
                          Object {
                            "is_marketable": false,
                            "modes": Array [
                              "masters",
                            ],
                            "name": "Masters Only",
                            "uuid": "f394732f-ba43-4260-8ada-06a9f18e7160",
                          },
                        ],
                        "entitlement_types": Array [],
                        "name": "Masters Only",
                        "tracks": Array [
                          Object {
                            "mode": Object {
                              "certificate_type": "",
                              "is_credit_eligible": false,
                              "is_id_verified": false,
                              "name": "Masters",
                              "payee": "organization",
                              "slug": "masters",
                            },
                            "seat_type": null,
                          },
                        ],
                        "uuid": "7b41992e-f268-4331-8ba9-72acb0880454",
                      },
                      Object {
                        "course_run_types": Array [
                          Object {
                            "is_marketable": true,
                            "modes": Array [
                              "verified",
                              "audit",
                            ],
                            "name": "Verified and Audit",
                            "uuid": "4e260c57-24ef-46c1-9a0d-5ec3a30f6b0c",
                          },
                          Object {
                            "is_marketable": true,
                            "modes": Array [
                              "audit",
                            ],
                            "name": "Audit Only",
                            "uuid": "cfacfc62-54bd-4e1b-939a-5a94f12fbd8d",
                          },
                        ],
                        "entitlement_types": Array [
                          "verified",
                        ],
                        "name": "Verified and Audit",
                        "tracks": Array [
                          Object {
                            "mode": Object {
                              "certificate_type": "verified",
                              "is_credit_eligible": false,
                              "is_id_verified": true,
                              "name": "Verified",
                              "payee": "platform",
                              "slug": "verified",
                            },
                            "seat_type": Object {
                              "name": "Verified",
                              "slug": "verified",
                            },
                          },
                          Object {
                            "mode": Object {
                              "certificate_type": "",
                              "is_credit_eligible": false,
                              "is_id_verified": false,
                              "name": "Audit",
                              "payee": "",
                              "slug": "audit",
                            },
                            "seat_type": Object {
                              "name": "Audit",
                              "slug": "audit",
                            },
                          },
                        ],
                        "uuid": "8a8f30e1-23ce-4ed3-a361-1325c656b67b",
                      },
                      Object {
                        "course_run_types": Array [
                          Object {
                            "is_marketable": true,
                            "modes": Array [
                              "professional",
                            ],
                            "name": "Professional Only",
                            "uuid": "6cebfc50-abca-4674-b4c7-249b31af6ddb",
                          },
                        ],
                        "entitlement_types": Array [
                          "professional",
                        ],
                        "name": "Professional Only",
                        "tracks": Array [
                          Object {
                            "mode": Object {
                              "certificate_type": "professional",
                              "is_credit_eligible": false,
                              "is_id_verified": true,
                              "name": "Professional",
                              "payee": "platform",
                              "slug": "professional",
                            },
                            "seat_type": Object {
                              "name": "Professional",
                              "slug": "professional",
                            },
                          },
                        ],
                        "uuid": "fdde7d04-7ce0-4ff7-ac90-7f518e90ac1e",
                      },
                      Object {
                        "course_run_types": Array [
                          Object {
                            "is_marketable": true,
                            "modes": Array [
                              "audit",
                            ],
                            "name": "Audit Only",
                            "uuid": "cfacfc62-54bd-4e1b-939a-5a94f12fbd8d",
                          },
                        ],
                        "entitlement_types": Array [
                          "audit",
                        ],
                        "name": "Audit Only",
                        "tracks": Array [
                          Object {
                            "mode": Object {
                              "certificate_type": "",
                              "is_credit_eligible": false,
                              "is_id_verified": false,
                              "name": "Audit",
                              "payee": "",
                              "slug": "audit",
                            },
                            "seat_type": Object {
                              "name": "Audit",
                              "slug": "audit",
                            },
                          },
                        ],
                        "uuid": "03e09c15-4127-4031-bc02-e20fcbdf09f0",
                      },
                    ],
                  },
                },
              },
            },
            "error": null,
            "isFetching": false,
          }
        }
        courseRunLabels={Array []}
        courseRunOptions={Object {}}
        currentFormValues={Object {}}
        destroyOnUnmount={true}
        enableReinitialize={false}
        forceUnregisterOnUnmount={false}
        form="create-course-run-form"
        getFormState={[Function]}
        initialValues={
          Object {
            "rerun": "",
            "run_type": undefined,
          }
        }
        isCreating={false}
        keepDirtyOnReinitialize={false}
        onSubmit={[Function]}
        persistentSubmitErrors={false}
        pure={true}
        shouldAsyncValidate={[Function]}
        shouldError={[Function]}
        shouldValidate={[Function]}
        shouldWarn={[Function]}
        submitAsSideEffect={false}
        title=""
        touchOnBlur={true}
        touchOnChange={false}
        updateUnregisteredFields={false}
        uuid=""
      />
    </div>
  </PageContainer>
</Fragment>
`;

exports[`CreateCourseRunPage renders html correctly when creating 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    Create Course Run
  </HelmetWrapper>
  <PageContainer
    className=""
    wide={false}
  >
    <div>
      <ReduxForm
        courseOptions={Object {}}
        courseRunLabels={Array []}
        courseRunOptions={Object {}}
        courseRunTypeOptions={Array []}
        currentFormValues={Object {}}
        destroyOnUnmount={true}
        enableReinitialize={false}
        forceUnregisterOnUnmount={false}
        form="create-course-run-form"
        getFormState={[Function]}
        initialValues={
          Object {
            "rerun": "",
            "run_type": undefined,
          }
        }
        isCreating={true}
        keepDirtyOnReinitialize={false}
        onSubmit={[Function]}
        persistentSubmitErrors={false}
        pure={true}
        shouldAsyncValidate={[Function]}
        shouldError={[Function]}
        shouldValidate={[Function]}
        shouldWarn={[Function]}
        submitAsSideEffect={false}
        title=""
        touchOnBlur={true}
        touchOnChange={false}
        updateUnregisteredFields={false}
        uuid=""
      />
    </div>
  </PageContainer>
</Fragment>
`;

exports[`CreateCourseRunPage renders html correctly when error 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    Create Course Run
  </HelmetWrapper>
  <PageContainer
    className=""
    wide={false}
  >
    <div>
      <ReduxForm
        courseOptions={Object {}}
        courseRunLabels={Array []}
        courseRunOptions={Object {}}
        courseRunTypeOptions={Array []}
        currentFormValues={Object {}}
        destroyOnUnmount={true}
        enableReinitialize={false}
        forceUnregisterOnUnmount={false}
        form="create-course-run-form"
        getFormState={[Function]}
        initialValues={
          Object {
            "rerun": "",
            "run_type": undefined,
          }
        }
        isCreating={false}
        keepDirtyOnReinitialize={false}
        onSubmit={[Function]}
        persistentSubmitErrors={false}
        pure={true}
        shouldAsyncValidate={[Function]}
        shouldError={[Function]}
        shouldValidate={[Function]}
        shouldWarn={[Function]}
        submitAsSideEffect={false}
        title=""
        touchOnBlur={true}
        touchOnChange={false}
        updateUnregisteredFields={false}
        uuid=""
      />
      <StatusAlert
        alertType="danger"
        className="mt-3"
        dismissible={false}
        iconClassNames={Array []}
        id="create-error"
        message={
          Array [
            "failed",
            <br />,
          ]
        }
        onClose={[Function]}
        title={null}
      />
    </div>
  </PageContainer>
</Fragment>
`;

exports[`CreateCourseRunPage renders html correctly when fetching 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    Create Course Run
  </HelmetWrapper>
  <PageContainer
    className=""
    wide={false}
  >
    <LoadingSpinner
      message="Loading…"
    />
  </PageContainer>
</Fragment>
`;

exports[`CreateCourseRunPage renders html correctly with Course Type 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    Create Course Run
  </HelmetWrapper>
  <PageContainer
    className=""
    wide={false}
  >
    <div>
      <ReduxForm
        courseOptions={
          Object {
            "data": Object {
              "actions": Object {
                "POST": Object {
                  "level_type": Object {
                    "choices": Array [
                      Object {
                        "display_name": "Beginner",
                        "value": "beginner",
                      },
                      Object {
                        "display_name": "Intermediate",
                        "value": "intermediate",
                      },
                      Object {
                        "display_name": "Advanced",
                        "value": "advanced",
                      },
                    ],
                  },
                  "subjects": Object {
                    "child": Object {
                      "choices": Array [
                        Object {
                          "display_name": "Business",
                          "value": "business",
                        },
                        Object {
                          "display_name": "Chemistry",
                          "value": "chemistry",
                        },
                        Object {
                          "display_name": "English",
                          "value": "english",
                        },
                        Object {
                          "display_name": "Security",
                          "value": "security",
                        },
                      ],
                    },
                  },
                  "type": Object {
                    "type_options": Array [
                      Object {
                        "course_run_types": Array [
                          Object {
                            "is_marketable": true,
                            "modes": Array [
                              "credit",
                              "verified",
                              "audit",
                            ],
                            "name": "Credit",
                            "uuid": "f17e29d6-4648-4bb5-a199-97dc40f904aa",
                          },
                          Object {
                            "is_marketable": true,
                            "modes": Array [
                              "verified",
                              "audit",
                            ],
                            "name": "Verified and Audit",
                            "uuid": "4e260c57-24ef-46c1-9a0d-5ec3a30f6b0c",
                          },
                          Object {
                            "is_marketable": true,
                            "modes": Array [
                              "audit",
                            ],
                            "name": "Audit Only",
                            "uuid": "cfacfc62-54bd-4e1b-939a-5a94f12fbd8d",
                          },
                        ],
                        "entitlement_types": Array [
                          "verified",
                        ],
                        "name": "Credit",
                        "tracks": Array [
                          Object {
                            "mode": Object {
                              "certificate_type": "credit",
                              "is_credit_eligible": true,
                              "is_id_verified": true,
                              "name": "Credit",
                              "payee": "platform",
                              "slug": "credit",
                            },
                            "seat_type": Object {
                              "name": "Credit",
                              "slug": "credit",
                            },
                          },
                          Object {
                            "mode": Object {
                              "certificate_type": "verified",
                              "is_credit_eligible": false,
                              "is_id_verified": true,
                              "name": "Verified",
                              "payee": "platform",
                              "slug": "verified",
                            },
                            "seat_type": Object {
                              "name": "Verified",
                              "slug": "verified",
                            },
                          },
                          Object {
                            "mode": Object {
                              "certificate_type": "",
                              "is_credit_eligible": false,
                              "is_id_verified": false,
                              "name": "Audit",
                              "payee": "",
                              "slug": "audit",
                            },
                            "seat_type": Object {
                              "name": "Audit",
                              "slug": "audit",
                            },
                          },
                        ],
                        "uuid": "9521aa7d-801b-4a67-92c3-716ea30f5086",
                      },
                      Object {
                        "course_run_types": Array [
                          Object {
                            "is_marketable": false,
                            "modes": Array [
                              "masters",
                            ],
                            "name": "Masters Only",
                            "uuid": "f394732f-ba43-4260-8ada-06a9f18e7160",
                          },
                        ],
                        "entitlement_types": Array [],
                        "name": "Masters Only",
                        "tracks": Array [
                          Object {
                            "mode": Object {
                              "certificate_type": "",
                              "is_credit_eligible": false,
                              "is_id_verified": false,
                              "name": "Masters",
                              "payee": "organization",
                              "slug": "masters",
                            },
                            "seat_type": null,
                          },
                        ],
                        "uuid": "7b41992e-f268-4331-8ba9-72acb0880454",
                      },
                      Object {
                        "course_run_types": Array [
                          Object {
                            "is_marketable": true,
                            "modes": Array [
                              "verified",
                              "audit",
                            ],
                            "name": "Verified and Audit",
                            "uuid": "4e260c57-24ef-46c1-9a0d-5ec3a30f6b0c",
                          },
                          Object {
                            "is_marketable": true,
                            "modes": Array [
                              "audit",
                            ],
                            "name": "Audit Only",
                            "uuid": "cfacfc62-54bd-4e1b-939a-5a94f12fbd8d",
                          },
                        ],
                        "entitlement_types": Array [
                          "verified",
                        ],
                        "name": "Verified and Audit",
                        "tracks": Array [
                          Object {
                            "mode": Object {
                              "certificate_type": "verified",
                              "is_credit_eligible": false,
                              "is_id_verified": true,
                              "name": "Verified",
                              "payee": "platform",
                              "slug": "verified",
                            },
                            "seat_type": Object {
                              "name": "Verified",
                              "slug": "verified",
                            },
                          },
                          Object {
                            "mode": Object {
                              "certificate_type": "",
                              "is_credit_eligible": false,
                              "is_id_verified": false,
                              "name": "Audit",
                              "payee": "",
                              "slug": "audit",
                            },
                            "seat_type": Object {
                              "name": "Audit",
                              "slug": "audit",
                            },
                          },
                        ],
                        "uuid": "8a8f30e1-23ce-4ed3-a361-1325c656b67b",
                      },
                      Object {
                        "course_run_types": Array [
                          Object {
                            "is_marketable": true,
                            "modes": Array [
                              "professional",
                            ],
                            "name": "Professional Only",
                            "uuid": "6cebfc50-abca-4674-b4c7-249b31af6ddb",
                          },
                        ],
                        "entitlement_types": Array [
                          "professional",
                        ],
                        "name": "Professional Only",
                        "tracks": Array [
                          Object {
                            "mode": Object {
                              "certificate_type": "professional",
                              "is_credit_eligible": false,
                              "is_id_verified": true,
                              "name": "Professional",
                              "payee": "platform",
                              "slug": "professional",
                            },
                            "seat_type": Object {
                              "name": "Professional",
                              "slug": "professional",
                            },
                          },
                        ],
                        "uuid": "fdde7d04-7ce0-4ff7-ac90-7f518e90ac1e",
                      },
                      Object {
                        "course_run_types": Array [
                          Object {
                            "is_marketable": true,
                            "modes": Array [
                              "audit",
                            ],
                            "name": "Audit Only",
                            "uuid": "cfacfc62-54bd-4e1b-939a-5a94f12fbd8d",
                          },
                        ],
                        "entitlement_types": Array [
                          "audit",
                        ],
                        "name": "Audit Only",
                        "tracks": Array [
                          Object {
                            "mode": Object {
                              "certificate_type": "",
                              "is_credit_eligible": false,
                              "is_id_verified": false,
                              "name": "Audit",
                              "payee": "",
                              "slug": "audit",
                            },
                            "seat_type": Object {
                              "name": "Audit",
                              "slug": "audit",
                            },
                          },
                        ],
                        "uuid": "03e09c15-4127-4031-bc02-e20fcbdf09f0",
                      },
                    ],
                  },
                },
              },
            },
            "error": null,
            "isFetching": false,
          }
        }
        courseRunLabels={Array []}
        courseRunOptions={Object {}}
        courseRunTypeOptions={
          Array [
            Object {
              "label": "Select enrollment track",
              "value": "",
            },
            Object {
              "label": "Verified and Audit",
              "value": "4e260c57-24ef-46c1-9a0d-5ec3a30f6b0c",
            },
            Object {
              "label": "Audit Only",
              "value": "cfacfc62-54bd-4e1b-939a-5a94f12fbd8d",
            },
          ]
        }
        courseTypeUuid="8a8f30e1-23ce-4ed3-a361-1325c656b67b"
        currentFormValues={Object {}}
        destroyOnUnmount={true}
        enableReinitialize={false}
        forceUnregisterOnUnmount={false}
        form="create-course-run-form"
        getFormState={[Function]}
        initialValues={
          Object {
            "rerun": "",
            "run_type": "4e260c57-24ef-46c1-9a0d-5ec3a30f6b0c",
          }
        }
        isCreating={false}
        keepDirtyOnReinitialize={false}
        onSubmit={[Function]}
        persistentSubmitErrors={false}
        pure={true}
        shouldAsyncValidate={[Function]}
        shouldError={[Function]}
        shouldValidate={[Function]}
        shouldWarn={[Function]}
        submitAsSideEffect={false}
        title=""
        touchOnBlur={true}
        touchOnChange={false}
        updateUnregisteredFields={false}
        uuid=""
      />
    </div>
  </PageContainer>
</Fragment>
`;
